File: mCRO/func_mon_std.m
function X_mon_std=func_mon_std(X,dt)
%dt
%if matches("monthly",dt)
    dt=1.0;
%end 

X=reshape(X,[length(X),1]);

X_mon=reshape(X,12/dt,size(X,1)/(12/dt)); % calendar month x year x ensemble 
X_mon_std=std(X_mon,[],2);

end 

File: CRO/pyCRO/func_mon_std.py
import numpy as np

def func_mon_std(X, dt=1.0):
    X = np.array(X).flatten()
    months_per_year = int(12 / dt)
    num_years = int(len(X) / months_per_year)
    
    if len(X) % months_per_year != 0:
        X = X[:months_per_year * num_years]  # trim excess
    
    X_mon = X.reshape((num_years, months_per_year))
    X_mon_std = np.std(X_mon, axis=0)
    return X_mon_std
